### PR TITLE
Make shapes drawing more flexible

### PR DESCRIPTION
**Pull request by corot**
- Make vertices attribute changeable so we can alter the shape dynamically.
- Undo transform on GL10 after drawing, so next draws in current iteration don't get affected by it can draw (allows drawing more than one shape at a time)

### CODE REVIEW

**Comment by corot** [comment]
Btw, I need this for a new app to annotate maps with semantic information: annotations shapes can change depending on user actions and I draw a list of then with different tf on every draw call.

https://github.com/robotics-in-concert/rocon_android_apps/tree/hydro-devel/map_annotation

File: android_honeycomb_mr2/src/main/java/org/ros/android/view/visualization/shape/TriangleFanShape.java [hunk 1]
```diff
@@ -53,5 +54,10 @@ public TriangleFanShape(float[] vertices, Color color) {
   public void draw(GL10 gl) {
     super.draw(gl);
     Vertices.drawTriangleFan(gl, vertices, getColor());
+    OpenGlTransform.apply(gl, getTransform().invert());
```
Comment (damonkohler) [inline android_honeycomb_mr2/src/main/java/org/ros/android/view/visualization/shape/TriangleFanShape.java:57]: You're breaking the class contract by applying a transform in the draw method.

File: android_honeycomb_mr2/src/main/java/org/ros/android/view/visualization/shape/TriangleFanShape.java [hunk 2]
```diff
@@ -53,5 +54,10 @@ public TriangleFanShape(float[] vertices, Color color) {
...
+  }
+
+  protected void setVertices(float[] vertices) {
+    this.vertices = Vertices.toFloatBuffer(vertices);
```
Comment (damonkohler) [inline android_honeycomb_mr2/src/main/java/org/ros/android/view/visualization/shape/TriangleFanShape.java:61]: Why modify vertices? Shapes are intended to be immutable. Just create a new shape instead.

Comment (corot) [inline android_honeycomb_mr2/src/main/java/org/ros/android/view/visualization/shape/TriangleFanShape.java:61]: Yep, I accept is not a net solution... I feel a bit embarrassed! So I'll try better:
- tf: what about adding a reset mechanism to OpenGlTransform? I will call it after every shape drawing, so successive transforms don't accumulate.
- vertices: I only change the size of the shape, so first I tried to use scaling. But I didn't find an easy way to match a physical size (in meters) to gl scaling, so I went to the radical surgery procedure of changing vertices. Any suggestions?

I'm with another app just now, but I'll come back on this soon.

**Comment by damonkohler** [comment]
Take a look at #184. I think it probably takes care of your issue.

**Comment by corot** [comment]
Right! I did the code before this change. Thanks.
I will try to use scaling instead of changing vertices.

**Comment by corot** [comment]
mmm.... after getting a bit more experience with Android and rosjava, use gl scaling for my purposes turned to be so ridiculously easy that made me feel embarrassed. Sorry for make you guys waste time with this pull request.

**Pull request closed without merging**